presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -348,8 +349,9 @@ public LocalExecutionPlan plan(
         Set<Symbol> partitioningColumns = partitioningScheme.getPartitioning().getColumns();
 
         // partitioningColumns expected to have one column in the normal case, and zero columns when partitioning on a constant
-        checkArgument(!partitioningScheme.isReplicateNulls() || partitioningColumns.size() <= 1);
-        if (partitioningScheme.isReplicateNulls() && partitioningColumns.size() == 1) {
+        boolean isReplicateNulls = isReplicateNulls(partitioningScheme.getReplication());
+        checkArgument(!isReplicateNulls || partitioningColumns.size() <= 1);
+        if (isReplicateNulls && partitioningColumns.size() == 1) {
```

Comment:
Not sure. `isCompatibleWithColumns` is generic, implementation agnostic. Here you have code assertion ensuring `getOnlyElement(partitioningColumns)` is the right way to do.